### ISSUE TITLE
Issue with bandwidth

### ISSUE THREAD

**Issue by pkochubey** [tus/tus-js-client#133]
Hi. I have problem with bandwidth on any latest browser.
On client I use latest version tus-js-client, on server pure implementation tusd.

What I have:
- 1 Server on SSD
- 1 Client PC on SSD
- 1 Gb network

When I upload 1 file from client to server I have speed around 700Mb. When I upload 2-3 files I have speed 1Gb.
Why when I upload 1 file I can't receive upload speed 1Gb?

--- ^ all right this is specific environment. but if we create more lightweight version, all will be the same ---

- 1 Server on SSD, Network IN 500 Mb\s
- 1 Client PC on SSD Network Out 200 Mb\s

When I upload 1 file from client to server I have speed around 100Mb. When I upload 2-3 files I have speed 200Mb.
WTF? this is reproduced on Linux and Windows, on Chrome and Firefox

**Comment by Acconut** [comment]
> on server pure implementation tusd

What data storage are you using for tusd? Disk based, AWS S3 or GCS? Could you provide the command that you use to invoke tusd?

The only thing limiting tusd uploads when using a disk based storage is the write speed of your disk. What is your write speed for the SSD?

**Comment by pkochubey** [comment]
> What data storage are you using for tusd?
Disk

> Could you provide the command that you use to invoke tusd?
I use systemd for start service:
ExecStart=/bin/tusd_linux_amd64/tusd -dir /storage/tmp -port 10081 -base-path /tusd/ -hooks-http https://${BASE_DOMAIN}/api/hooks/tus

> What is your write speed for the SSD?
```
dd if=/dev/zero of=./largefile bs=1M count=1024
1024+0 records in
1024+0 records out
1073741824 bytes (1.1 GB, 1.0 GiB) copied, 0.795848 s, 1.3 GB/s
```

I think what problem not in storage speed, because when server and client in one network  I can push one file with speed 700 mb\s when client has long distance I can push one file only with speed 80-90 mb\s but bandwidth on remoted client 200 mb\s and yes I can use all this bandwidth if I will upload in one time 2-3 file, but when I upload 1 file I can't use the entire Internet channel for data transmission.

**Comment by pkochubey** [comment]
My conclusion - this problem with limitation speed for one xhr request. need implement concatenation extension into client tus for solved this issue.

**Comment by Acconut** [comment]
The browser might be one limitation, that's right. I will spend more time investigating this in the future but currently I cannot devote much time on that.

**Comment by fidelix** [comment]
Check this out:
https://github.com/tus/tus-js-client/issues/55

**Comment by Acconut** [comment]
Parallel upload requests will be implemented in tus-js-client v2: #167
